### PR TITLE
feat: refactored some bridges to avoid leaking resources part.2

### PR DESCRIPTION
**Pull request by HJianBo**
Fixes https://emqx.atlassian.net/browse/EMQX-9933

### <samp>🤖 Generated by Copilot at fc82c14</samp>

This pull request adds resource management for the pool names used by various bridge connectors using the `emqx_resource` module. It also improves the robustness of the `on_stop/2` functions of the connectors by using the resource registry instead of the state map. Additionally, it updates the version numbers and the change log of the affected bridge applications.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- ~Added tests for the changes~
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [x] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [x] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge_cassandra/src/emqx_bridge_cassandra_connector.erl [hunk 1]
```diff
@@ -132,6 +132,7 @@ on_start(
                 []
         end,
     State = parse_prepare_cql(Config),
+    ok = emqx_resource:allocate_resource(InstId, pool_name, InstId),
```
Comment (zmstone) [inline apps/emqx_bridge_cassandra/src/emqx_bridge_cassandra_connector.erl:135]: the pool name is constructed from InstId, so there should be no need to allocate resource ?

File: apps/emqx_bridge_clickhouse/src/emqx_bridge_clickhouse_connector.erl [hunk 2]
```diff
@@ -158,6 +158,7 @@ on_start(
             templates => Templates,
             connect_timeout => ConnectTimeout
         },
+        ok = emqx_resource:allocate_resource(InstanceID, pool_name, InstanceID),
```
Comment (zmstone) [inline apps/emqx_bridge_clickhouse/src/emqx_bridge_clickhouse_connector.erl:161]: the pool name is constructed from resource id?

**Review by zmstone — CHANGES_REQUESTED** [review]
please double check all the pool names, if a poo name is computed from reousrce id, there is no need to allocate anything.

File: changes/ce/feat-10895.en.md
Comment (JimMoen) [inline changes/ce/feat-10895.en.md:1]: maybe move to `changes/ee/` ?

File: apps/emqx_bridge_sqlserver/src/emqx_bridge_sqlserver_connector.erl [hunk 3]
```diff
@@ -215,6 +215,7 @@ on_start(
         sql_templates => parse_sql_template(Config),
         resource_opts => ResourceOpts
     },
+    ok = emqx_resource:allocate_resource(ResourceId, pool_name, PoolName),
```
Comment (JimMoen) [inline apps/emqx_bridge_sqlserver/src/emqx_bridge_sqlserver_connector.erl:218]: the `PoolName` is same as `ResourceId`, so there should be no need to allocate it.
simply calling `emqx_resource_pool:stop(ResourceId)` is enough.

**Review by zhongwencool — APPROVED** [review]

**Review by JimMoen — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**